packages/devtools_app/lib/src/memory/flutter/memory_filter.dart

```diff
@@ -0,0 +1,391 @@
+    String groupedName = libraryName;
+    bool hide = hideState;
+    if (isFiltered) {
+      if (filters.isDartLibraryName(libraryName)) {
+        groupedName = _prettyPrintDartAbbreviation;
+      } else if (filters.isFlutterLibraryName(libraryName)) {
+        groupedName = _prettyPrintFlutterAbbreviation;
+      }
+    }
+    hide = isFiltered;
+
+    // Used by checkboxes in dialog.
+    filteredGroup[groupedName] ??= [];
+    filteredGroup[groupedName].add(LibraryFilter(libraryName, hide));
+  }
+
+  void buildFilters() {
+    if (controller == null) return;
+
+    // First time filters created, populate with the default list of libraries
+    // to filters
+    if (controller.filteredLibrariesByGroupName.isEmpty) {
+      for (final library in controller.libraryFilters.librariesFiltered) {
+        addLibrary(library, true);
+      }
+      // If not snapshots, return no libraries to process.
+      if (controller.snapshots.isEmpty) return;
+    }
+
+    // No libraries to compute until a snapshot exist.
+    if (controller.snapshots.isEmpty) return;
+
+    final libraries = controller.computeAllLibraries().children;
+    libraries..sort((a, b) => a.name.compareTo(b.name));
+
+    for (final library in libraries) {
+      // Don't include external and filtered these are a composite and can't be filtered out.
+      if (library.name != externalLibraryName &&
+          library.name != filteredLibrariesName) {
+        addLibrary(library.name);
+      }
+    }
+  }
+
+  /// Process wildcard groups dart:* and package:flutter*. If a wildcard group is
+  /// toggled from on to off then all the dart: packages will appear if the group
+  /// dart:* is toggled back on then all the dart: packages must be removed.
+  void cleanupFilteredLibrariesByGroupName() {
+    final filteredGroup = controller.filteredLibrariesByGroupName;
+    final dartGlobal = filteredGroup[_prettyPrintDartAbbreviation].first.hide;
+    final flutterGlobal =
+        filteredGroup[_prettyPrintFlutterAbbreviation].first.hide;
+
+    filteredGroup.removeWhere((groupName, libraryFilter) {
+      if (dartGlobal &&
+          groupName != _prettyPrintDartAbbreviation &&
+          groupName.startsWith(_dartLibraryUriPrefix)) {
+        return true;
+      } else if (flutterGlobal &&
+          groupName != _prettyPrintFlutterAbbreviation &&
+          groupName.startsWith(_flutterLibraryUriPrefix)) {
+        return true;
+      }
+
+      return false;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    buildFilters();
+
+    return Dialog(
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20.0)),
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          return Container(
+            width: MediaQuery.of(context).size.width / 3,
+            height: constraints.maxHeight < 500
+                ? constraints.maxHeight
+                : constraints.maxHeight / 3 + 300,
```

Comment:
document why `/3 + 300` is a good value here.
It probably makes sense to choose a value so that 500 (or the threshold) would have the same height regardless. 
So for example,
`constraints.maxHeight / 3 + (500*2/3)`

Reply:
// Dialog has three main vertical sections:
    //      - three checkboxes
    //      - one list of libraries with at least 5 entries
    //      - one row of buttons Ok/Cancel

I think your computation is better plus I adjusted for smaller ranges.

            width: MediaQuery.of(context).size.width / 3,
            height: constraints.maxHeight < 400
                ? constraints.maxHeight
                : constraints.maxHeight / 3 + (400 * .7),


Still think there needs to be UX cleanup but I'll await a UX review.